staffing/views.py

```diff
@@ -1018,7 +1021,11 @@ def all_timesheet(request, year=None, month=None):
     previous_date = (month - timedelta(days=5)).replace(day=1)
     next_date = nextMonth(month)
 
-    timesheets = Timesheet.objects.filter(working_date__gte=month)  # Filter on current month
+    if subsidiary:
+        timesheets = Timesheet.objects.filter(working_date__gte=month, consultant__company=subsidiary)
+    else:
+        timesheets = Timesheet.objects.filter(working_date__gte=month)  # Filter on current month
```

Comment:
this line should be before the if subsidiary to avoid repetition of month filtering. And the if subsidiary block should only append filter on subsidiary.